### PR TITLE
49 subissue 25 create login use case

### LINKED ISSUE #49: [SubIssue 2.5] Create Login Use Case

### PR DESCRIPTION
**Pull request by dadler83**
New Features:
- login use case connected
- InMemoryUser class created and connected to use cases
- Main class extracted from login UI class
- Unnecessary classes in User Registration use case removed (i.e. void use case because no presenters are needed)

### CODE REVIEW

File: src/main/java/Main.java [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+import entity_layer.UserFactory;
+import use_cases.login.LoginController;
+import use_cases.login.LoginInputBoundary;
+import use_cases.login.LoginInteractor;
+import use_cases.user_registration.UserRegistrationController;
 import use_cases.user_registration.UserRegistrationInteractor;
 
 import javax.swing.*;
 
 public class Main {
```
Comment (MMachadoUofT) [inline src/main/java/Main.java:16]: Thanks so much for setting this up! Extracting the login into a Main driver will definitely help clean up our architecture and make the dependencies a lot more fluid.

**Review by MMachadoUofT — APPROVED** [review]
This is incredible! You've done excellent work on this project. Admittedly, a lot of the cryptographic work is out of my scope of understanding, but your code is still highly legible and makes it very easy to follow along! The inMemoryUser class that we discussed making is also going to be a major plus in the functionality of our interactors. 

It's also good that we now officially have a driver Main class; with the original implementation of having this be the LogIn use case, we definitely would've run into problems in setting up our dependencies. Extracting this will definitely prove useful in the long run.

**Review by bmguiler — APPROVED** [review]
Thank you for making the InMemoryUser class, this will help everyone who is making schedule / task use cases to be able to follow clean architecture

**Review by Eggathin — APPROVED** [review]
Excellent work, will use this as reference for my part of the project

**Pull request merged**